miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java

```diff
@@ -436,16 +440,14 @@ public int compare(Lab o1, Lab o2) {
   @Autowired
   private SamplePurposeService samplePurposeService;
 
-  public List<SamplePurpose> getSamplePurposes(SampleAliquot sample) throws IOException {
-    List<SamplePurpose> list = new ArrayList<>(samplePurposeService.getAll());
-    Collections.sort(list, new Comparator<SamplePurpose>() {
-      @Override
-      public int compare(SamplePurpose o1, SamplePurpose o2) {
-        return o1.getAlias().compareTo(o2.getAlias());
-      }
-    });
-    return list.stream().filter(samPurpose -> !samPurpose.isArchived() ||
-        (sample.getSamplePurpose() != null && sample.getSamplePurpose().getId() == samPurpose.getId()))
+  private static final Comparator<Aliasable> ALIAS_COMPARATOR = (o1, o2) -> o1.getAlias().compareTo(o2.getAlias());
```

Comment:
You can also `Comparator.comparing(Aliasable::getAlias)`

Reply:
Ah, I thought there was something like that somewhere... Thanks!